iml-gui/crate/src/page/volumes.rs

```diff
@@ -65,15 +66,24 @@ impl RecordChange<Msg> for Model {
         }
     }
     fn set_records(&mut self, cache: &ArcCache, orders: &mut impl Orders<Msg, GMsg>) {
-        self.rows = build_rows(cache);
+        self.rows = build_rows(cache, &self.host);
 
         orders.proxy(Msg::Page).send_msg(paging::Msg::SetTotal(self.rows.len()));
 
         orders.send_msg(Msg::Sort);
     }
 }
 
-fn build_rows(cache: &ArcCache) -> Vec<Row> {
+impl From<&Arc<Host>> for Model {
+    fn from(host: &Arc<Host>) -> Self {
+        Self {
+            host: Some(host.clone()),
```

Comment:
```suggestion
            host: Some(Arc::clone(host)),
```

It's worth explicitly doing the clone this way to show we are not actually cloning the Host, but instead incrementing a refcount